### PR TITLE
Move transformers to their own cpp/h files

### PR DESCRIPTION
**Pull request by mmcgr**
Move transformers out of non-transformer files

AstPragmaChecker also contained a non-transformer check of souffle parameters which has been moved to main.cpp

DebugReport/DebugReporter were a little confused. As well as moving, the transformer responsibilities has been shuffled a bit so that the transformer is not responsible for non-transformer duties.

ComponentModel contained an AstAnalysis which has been moved to an Analysis file name.

### CODE REVIEW

**Comment by mmcgr** [comment]
There are still several transformers that have cpp files, but one shared header file with the class declarations. I've left that for a later PR with a move to subdirectories when the structure will be clearer.

File: src/AstPragma.h [hunk 1]
```diff
@@ -17,17 +17,13 @@
...
 #include <utility>
-#include <vector>
 
 namespace souffle {
 
-class AstTranslationUnit;
-
 /**
  * @class AstPragma
  * @brief Representation of a global option
```
Comment (b-scholz) [inline src/AstPragma.h]: Are we sure we need the default constructor?

**Comment by mmcgr** [comment]
I do agree about DebugReport. I split it up, and at least we don't have RAM using an AstTransformer any more, but it still needs more work.

File: src/AstPragma.h (hunk 1 above)
Comment (mmcgr) [reply src/AstPragma.h]: It's only used by the clone method, so I refactored it out.

**Comment by mmcgr** [comment]
Partial fix of #1255

**Pull request merged**